source/common/grpc/async_client_impl.h

```diff
@@ -82,7 +80,8 @@ class AsyncStreamImpl : public AsyncStream<RequestType>,
                   const Protobuf::MethodDescriptor& service_method,
                   AsyncStreamCallbacks<ResponseType>& callbacks,
                   const Optional<std::chrono::milliseconds>& timeout)
-      : parent_(parent), service_method_(service_method), callbacks_(callbacks), timeout_(timeout) {
+      : dispatcher_(parent.cm_.httpAsyncClientForCluster(parent.remote_cluster_name_).dispatcher()),
```

Comment:
perf nit. cm_.httpAsyncClientForCluster() will end up doing TLS lookup which is not free. Can we either grab the dispatcher off the client in initialize() (and just store in pointer) or potentially change interface so we can grab dispatcher off of stream vs. client?